jobs/regular/check_akismet_post.rb

```diff
@@ -9,7 +9,7 @@ def execute(args)
       return unless SiteSetting.akismet_enabled?
 
       post = Post.find_by(id: args[:post_id], user_deleted: false)
-      raise Discourse::InvalidParameters.new(:post_id) if post.nil?
+      return if post.blank?
```

Comment:
This can also be a one-liner

```suggestion
      return unless post = Post.find_by(id: args[:post_id], user_deleted: false)
```